lib/simulator-xcode-8.js

```diff
@@ -33,94 +37,89 @@ class SimulatorXcode8 extends SimulatorXcode7 {
     }
   }
 
-  get startupTimeout () {
-    return STARTUP_TIMEOUT;
+  get startupPollBundleId () {
+    return SPRINGBOARD_BUNDLE_ID;
   }
 
-  get startupPollCommand () {
-    return {
-      cmd: 'bash',
-      args: [
-        '-c',
-        'ps axo command | grep Simulator | grep nsurlstoraged | grep -v bash | grep -v grep'
-      ],
-    };
+  get startupTimeout () {
+    return STARTUP_TIMEOUT;
   }
 
   async waitForBoot (startupTimeout) {
-    // wait for the simulator to boot
-    // waiting for the simulator status to be 'booted' isn't good enough
-    // it claims to be booted way before finishing loading
-    // let's wait for the magic nsurlstoraged process, which signals the booting has been completed
     const startupTimestamp = process.hrtime();
-    let {cmd, args} = this.startupPollCommand;
+    let lastError = null;
     try {
       let isOnBootCompletedEmitted = false;
       await waitForCondition(async () => {
         try {
-          let {stdout} = await exec(cmd, args);
-          if (stdout.trim().length > 0) {
+          // 'springboard' process should be the last one to start after boot
+          // 'simctl launch' will block until this process is running or fail if booting is still in progress
+          const {stdout} = await exec('xcrun', ['simctl', 'launch', this.udid, this.startupPollBundleId]);
+          if (PROCESS_LAUNCH_OK_PATTERN(this.startupPollBundleId).test(stdout)) {
             if (!isOnBootCompletedEmitted) {
               isOnBootCompletedEmitted = true;
               this.emit(BOOT_COMPLETED_EVENT);
             }
-            // 'springboard' process should be the last one to start after boot
-            // 'simctl launch' will block until this process is running
-            await exec('xcrun', ['simctl', 'launch', this.udid, 'com.apple.springboard']);
+
             return true;
           }
-        } catch (ign) {
-          // Continue iteration in case of error
+        } catch (err) {
+          lastError = err.stderr || err.message;
         }
         return false;
       }, {waitMs: startupTimeout, intervalMs: 500});
     } catch (err) {
-      log.errorAndThrow(`Simulator is not booted after ${process.hrtime(startupTimestamp)[0]} seconds`);
+      log.errorAndThrow(`Simulator is not booted after ${process.hrtime(startupTimestamp)[0]} seconds because of ` +
+                        `"${lastError ? '"' + lastError + '"' : 'an unknown error'}"`);
```

Comment:
The formatting here seems strange. Won't it print either `... because of ""whatever the error""` or `... because of "an unknown error"`?

Perhaps we need
```js
`${lastError ? `'${lastError}'` : 'an unknown error'}`);
```